aopy/preproc/quality.py

```diff
@@ -35,12 +37,77 @@ def bad_channel_detection(data, srate, lf_c=100., sg_win_t=8., sg_over_t=4., sg_
 
     psd_var = np.var(Sxx_low_psd,axis=0)
     norm_psd_var = psd_var/npla.norm(psd_var)
-    low_var_θ = np.mean(norm_psd_var)/3
+    low_var_θ = np.mean(norm_psd_var)/num_th
     bad_ch_mask = norm_psd_var <= low_var_θ
 
     return bad_ch_mask
 
 
+def screenBadECoGchannels(data, th=0.05, numsd=5.0, debug=False, verbose=True):
+    '''
+    Detect badchannels. This code is originally Amy's code in pesaran lab.
+    This function has 2 steps. In the 1st step, it detects tentative bad channels whose SD is less than th or more than 1-th in CDF
+    In the 2nd step, it extracts bad channels from tentative bad channels by finding conditions where (SD - median of SD) is outside numsd*SD[~badch]
+    
+    Args:
+        data (nt, nch): neural data
+        th (float, optional): threshold to detect bad channels for 1st screening
+        numsd (float, optional): number of standard deviations above zero to detect bad channels for 2nd screening
+        debug (bool, optional): if True, display a figure showing the threshold crossings
+        verbose (bool, optional): if True, print bad channels
+        
+    Returns:
+        bad_ch (nch) : logical array indicating bad channels
+
+    Examples:
+        
+        .. code-block:: python
+        
+        test_data = np.random.normal(10,0.5,(10000, 200))
+        test_data[0, 10] = 25
+        test_data[5, 150] = 30
+        bad_ch = screenBadECoGchannels(test_data, th=0.05, numsd=5.0, debug=True, verbose=False)
+        
+        .. image:: _images/screenBadECoGchannels.png
+    '''
+    
+    assert th > 0 and th < 1, "Threshold must be between 0 and 1"
+    assert numsd > 0, "numsd must be more than 0"
+    
+    sd = np.std(data, axis=0)
+    med_sd = np.median(sd)
+    
+    nbins = 10000
+    hist, bins = np.histogram(sd, nbins)
+    
+    CDF = np.cumsum(hist)/np.sum(hist)
+    bottom = bins[np.where(CDF<th)[0][-1]]
+    top = bins[np.where(CDF>1-th)[0][0]]
+    bad_ch = (sd < bottom) | (sd > top)
+    
+    inrange = np.abs(sd - med_sd) <= numsd*np.std(sd[~bad_ch], ddof=1)
+    bad_ch = bad_ch & ~inrange
+    
+    if debug:
```

Comment:
I like the debug figure, but it doesn't show anything about the second selection step. i would get rid of the middle subplot since it's entirely captured by the cdf plot, and maybe in the third subplot you could subtract the median sd from everything and then draw the `numsd*np.std(std[~bad_ch], ddof=1)` threshold, similar to the first subplot? then label them "screening 1" and "screening 2"